SimPEG/Directives.py

```diff
@@ -49,39 +49,56 @@ def opt(self):
     @property
     def reg(self):
         if getattr(self, '_reg', None) is None:
-            self._reg = self.invProb.reg
+            self.reg = self.invProb.reg  # go through the setter
         return self._reg
 
     @reg.setter
     def reg(self, value):
-        assert any([isinstance(value, regtype) for regtype in self._regPair]),(
+        assert any([isinstance(value, regtype) for regtype in self._regPair]), (
             "Regularization must be in {}, not {}".format(
                 self._regPair, type(value)
             )
         )
-        self._reg = reg
+
+        if isinstance(value, Regularization.BaseComboRegularization):
```

Comment:
we can do this in a separate pr, but we should add docstrings to `reg` `dmisfit` etc so we communicate that these will always return a combo objective function